src/libraries/Microsoft.Extensions.Primitives/tests/ChangeTokenTest.cs

```diff
@@ -234,6 +235,12 @@ public void DoubleDisposeDisposesOnce()
             Assert.Equal(2, provider.DisposeCalls);
         }
 
+        [Fact]
+        public void NullTokenDisposeShouldNotThrow()
+        {
+            ChangeToken.OnChange(() => null, () => Debug.Fail("Should never be called")).Dispose();
```

Comment:
I'd recommend to use Assert.True(false) instead - Debug.Fail will stop the entire process rather than failing a single test case